src/System Application/App/Azure AD Tenant/src/AzureADTenantImpl.Codeunit.al

```diff
@@ -15,24 +15,49 @@ codeunit 3705 "Azure AD Tenant Impl."
     InherentPermissions = X;
 
     var
+        AzureADGraph: Codeunit "Azure AD Graph";
+        TenantInfo: DotNet TenantInfo;
         NavTenantSettingsHelper: DotNet NavTenantSettingsHelper;
         TenantDomainNameErr: Label 'Failed to retrieve the Microsoft Entra tenant domain name.';
+        CountryLetterCodeErr: Label 'Failed to retrieve the Microsoft Entra tenant country letter code.';
+        PreferredLanguageErr: Label 'Failed to retrieve the Microsoft Entra tenant preferred language code.';
 
     procedure GetAadTenantId() TenantIdValue: Text
     begin
         NavTenantSettingsHelper.TryGetStringTenantSetting('AADTENANTID', TenantIdValue);
     end;
 
     procedure GetAadTenantDomainName(): Text;
-    var
-        AzureADGraph: Codeunit "Azure AD Graph";
-        TenantInfo: DotNet TenantInfo;
     begin
-        AzureADGraph.GetTenantDetail(TenantInfo);
+        Initialize();
         if not IsNull(TenantInfo) then
             exit(TenantInfo.InitialDomain());
 
         Error(TenantDomainNameErr);
     end;
+
+    procedure GetCountryLetterCode(): Text;
+    begin
+        Initialize();
+        if not IsNull(TenantInfo) then
+            exit(TenantInfo.CountryLetterCode());
+
+        Error(CountryLetterCodeErr);
+    end;
+
+    procedure GetPreferredLanguage(): Text;
```

Comment:
Do we know a maximum length for this language? Then we should specify it to facilitate storing it in a table.

Reply:
It should be 2 but there is no guarentee

Reply:
So how should I handle this if I want to store it in a table?

Reply:
Right now I have no idea.. I would need a blob?